src/QtHtmlReader.cpp

```diff
@@ -0,0 +1,267 @@
+		}
+
+		//set background
+		painter.setBackground(QBrush(background_color.c_str()));
+
+		//draw text
+		QTextDocument text_document;
+
+		//disable redo/undo stack as not needed
+		text_document.setUndoRedoEnabled(false);
+
+		//create the HTML/CSS document
+		text_document.setTextWidth(width);
+		text_document.setDefaultStyleSheet(css.c_str());
+		text_document.setHtml(html.c_str());
+
+		int td_height = text_document.documentLayout()->documentSize().height();
+
+ 		if (gravity == GRAVITY_TOP_LEFT || gravity == GRAVITY_TOP || gravity == GRAVITY_TOP_RIGHT) {
+ 			painter.translate(x_offset, y_offset);
+ 		} else if (gravity == GRAVITY_LEFT || gravity == GRAVITY_CENTER || gravity == GRAVITY_RIGHT) {
+ 			painter.translate(x_offset, (height - td_height) / 2 + y_offset);
+ 		} else if (gravity == GRAVITY_BOTTOM_LEFT || gravity == GRAVITY_BOTTOM_RIGHT || gravity == GRAVITY_BOTTOM) {
+ 			painter.translate(x_offset, height - td_height + y_offset);
+ 		}
+
+ 		if (gravity == GRAVITY_TOP_LEFT || gravity == GRAVITY_LEFT || gravity == GRAVITY_BOTTOM_LEFT) {
+ 			text_document.setDefaultTextOption(QTextOption(Qt::AlignLeft));
+ 		} else if (gravity == GRAVITY_CENTER || gravity == GRAVITY_TOP || gravity == GRAVITY_BOTTOM) {
+ 			text_document.setDefaultTextOption(QTextOption(Qt::AlignHCenter));
+ 		} else if (gravity == GRAVITY_TOP_RIGHT || gravity == GRAVITY_RIGHT|| gravity == GRAVITY_BOTTOM_RIGHT) {
+ 			text_document.setDefaultTextOption(QTextOption(Qt::AlignRight));
+ 		}
+
+ 		// Draw image
+		text_document.drawContents(&painter);
+
+		painter.end();
+
+		// Update image properties
+		info.has_audio = false;
+		info.has_video = true;
+		info.file_size = 0;
+		info.vcodec = "QImage";
+		info.width = width;
+		info.height = height;
+		info.pixel_ratio.num = 1;
+		info.pixel_ratio.den = 1;
+		info.duration = 60 * 60 * 24; // 24 hour duration
+		info.fps.num = 30;
+		info.fps.den = 1;
+		info.video_timebase.num = 1;
+		info.video_timebase.den = 30;
+		info.video_length = round(info.duration * info.fps.ToDouble());
+
+		// Calculate the DAR (display aspect ratio)
+		Fraction size(info.width * info.pixel_ratio.num, info.height * info.pixel_ratio.den);
+
+		// Reduce size fraction
+		size.Reduce();
+
+		// Set the ratio based on the reduced fraction
+		info.display_ratio.num = size.num;
+		info.display_ratio.den = size.den;
+
+		// Mark as "open"
+		is_open = true;
+	}
+}
+
+// Close reader
+void QtHtmlReader::Close()
+{
+	// Close all objects, if reader is 'open'
+	if (is_open)
+	{
+		// Mark as "closed"
+		is_open = false;
+
+		// Delete the image
+		image.reset();
+
+		info.vcodec = "";
+		info.acodec = "";
+	}
+}
+
+// Get an openshot::Frame object for a specific frame number of this reader.
+std::shared_ptr<Frame> QtHtmlReader::GetFrame(int64_t requested_frame)
+{
+	if (image)
+	{
+		// Create or get frame object
+		std::shared_ptr<Frame> image_frame(new Frame(requested_frame, image->size().width(), image->size().height(), background_color, 0, 2));
```

Comment:
Same here, this can be:
```cpp
std::shared_ptr<Frame> image_frame(new Frame(requested_frame, image->size().width(),
                                             image->size().height(), background_color));
```

Reply:
```suggestion
		std::shared_ptr<Frame> image_frame(new Frame(requested_frame, image->size().width(), image->size().height(), background_color));
```